src/py/mat3ra/made/tools/build/passivation/builders.py

```diff
@@ -0,0 +1,213 @@
+from typing import List
+
+import numpy as np
+from mat3ra.made.material import Material
+from pydantic import BaseModel
+
+from ...enums import SurfaceTypes
+from ...analyze import (
+    get_surface_atom_indices,
+    get_undercoordinated_atom_indices,
+    get_nearest_neighbors_atom_indices,
+    get_coordination_numbers,
+)
+from ...modify import translate_to_z_level
+from ...build import BaseBuilder
+from .configuration import (
+    PassivationConfiguration,
+)
+
+
+class PassivationBuilder(BaseBuilder):
+    """
+    Base class for passivation builders.
+    """
+
+    _GeneratedItemType = Material
+    _ConfigurationType = PassivationConfiguration
+
+    def _generate(self, configuration: BaseBuilder._ConfigurationType) -> List[Material]:
+        return [self.create_passivated_material(configuration)]
+
+    def _update_material_name(
+        self, material: BaseBuilder._GeneratedItemType, configuration: BaseBuilder._ConfigurationType
+    ) -> BaseBuilder._GeneratedItemType:
+        material = super()._update_material_name(material, configuration)
+        material.name += f" {configuration.passivant}-passivated"
+        return material
+
+    def create_passivated_material(self, configuration: BaseBuilder._ConfigurationType) -> Material:
+        material = translate_to_z_level(configuration.slab, "center")
+        return material
+
+    def _add_passivant_atoms(self, material: Material, coordinates: list, passivant: str) -> Material:
+        """
+        Add passivant atoms to the provided coordinates in the material.
+
+        Args:
+            material (Material): The material object to add passivant atoms to.
+            coordinates (list): The coordinates to add passivant atoms to.
+            passivant (str): The chemical symbol of the passivating atom (e.g., 'H').
+
+        Returns:
+            Material: The material object with passivation atoms added.
+        """
+        for coord in coordinates:
+            material.add_atom(passivant, coord)
+        return material
+
+
+class SurfacePassivationBuilderParameters(BaseModel):
+    """
+    Parameters for the SurfacePassivationBuilder.
+
+    Args:
+        shadowing_radius (float): Radius for atoms shadowing underlying from passivation, in Angstroms.
+        depth (float): Depth from the top to look for exposed surface atoms to passivate, in Angstroms.
+    """
+
+    shadowing_radius: float = 2.5
+    depth: float = 5.0
```

Comment:
Why do we need such a deep value here - 5.0A? What is the "top" - the top of the cell or the top layer of atoms??

Reply:
We need to explain what "top" means here